DatadogSessionReplay/Tests/Recorder/RecorderTests.swift

```diff
@@ -8,42 +8,32 @@ import XCTest
 @testable import DatadogSessionReplay
 @testable import TestUtilities
 
-private class RUMContextObserverMock: RUMContextObserver {
-    private var queue: Queue?
-    private var onNew: ((RUMContext?) -> Void)?
-
-    func observe(on queue: Queue, notify: @escaping (RUMContext?) -> Void) {
-        self.queue = queue
-        self.onNew = notify
-    }
-
-    func notify(rumContext: RUMContext?) {
-        queue?.run { self.onNew?(rumContext) }
-    }
-}
-
 class RecorderTests: XCTestCase {
+    private var recorder: Recorder?
+
     func testGivenRUMContextAvailable_whenStarted_itCapturesSnapshotsAndPassesThemToProcessor() {
         let numberOfSnapshots = 10
         let mockViewTreeSnapshots: [ViewTreeSnapshot] = .mockRandom(count: numberOfSnapshots)
         let mockTouchSnapshots: [TouchSnapshot] = .mockRandom(count: numberOfSnapshots)
         let rumContextObserver = RUMContextObserverMock()
         let processor = ProcessorSpy()
+        let scheduler = TestScheduler(numberOfRepeats: numberOfSnapshots)
 
         // Given
-        let recorder = Recorder(
+        recorder = Recorder(
             configuration: .mockAny(),
             rumContextObserver: rumContextObserver,
             uiApplicationSwizzler: .mockAny(),
-            scheduler: TestScheduler(numberOfRepeats: numberOfSnapshots),
+            scheduler: scheduler,
+            recordingCoordinator: RecordingCoordinationMock(currentRUMContext: .mockRandom()),
             viewTreeSnapshotProducer: ViewTreeSnapshotProducerMock(succeedingSnapshots: mockViewTreeSnapshots),
             touchSnapshotProducer: TouchSnapshotProducerMock(succeedingSnapshots: mockTouchSnapshots),
             snapshotProcessor: processor
         )
         rumContextObserver.notify(rumContext: .mockAny())
```

Comment:
change request/ This is not needed anymore, `Recorder` doesn't depend on `rumContextObserver`. This could be an argument towards the small refactor I suggested earlier - we loose clarity on what is required `Recorder` dependency.